include/bout/sys/expressionparser.hxx

```diff
@@ -194,7 +194,8 @@ private:
 
 class ParseException : public std::exception {
 public:
-  ParseException(const char *, ...);
+  ParseException(const char *, ...)
+    __attribute__ ((format (printf, 2, 3)));
```

Comment:
Isn't this `GCC` specific so may not work with any other compiler family?

Reply:
Yes, this is non-portable, although it seems clang and Intel both accept it. It should be wrapped up in a macro.

One day we'll actually move to `{fmt}` and we'll get proper type-checking

Reply:
Probably, but if we can fix them at least with one compiler, than this is fine.

Also, gcc is available on most systems.

I tagged it as WIP, so we can guard it/put it in a macro if it causes issues for other compilers.

Reply:
thoughts on the macro name?
`_BOUT_FORMAT(i,j)?

Reply:
We should avoid names with leading underscores followed by capitals. Maybe just `BOUT_FORMAT`, or `BOUT_FORMAT_ARGS`

Reply:
is master ok, or should it go into next only?